### ISSUE TITLE
`XdocsExamplesAstConsistencyTest` currently misses three classes of inconsistency between xdoc example files that share the same check configuration.

### ISSUE THREAD

**Issue by smita1078** [checkstyle/checkstyle#20284]
## Issue 1 — identifier names are not compared

In `StructuralAstNode.isLiteralToken()`, text is only preserved for numeric, string, boolean, and null literals. `IDENT` tokens always produce `text=null`, so two examples with structurally identical code but different variable, parameter, or annotation names compare as equal and the test passes silently.

**Example** (`checks/naming/lambdaparametername/`):

```java
// Example1
Function<String, String> function2 = S  -> S.trim();
// Example2
Function<String, String> function2 = _s -> _s.trim();
```

Both `S` and `_s` are `IDENT` nodes whose text is discarded — the test reports no mismatch.

Class and constructor names are intentionally excluded via `isClassOrConstructorName()` so that `Example1` vs `Example2` class name differences do not cause false positives. That guard is unaffected by this fix.

---

## Issue 2 — line numbers are not compared

The test compares token types, literal text, and children — but not source positions. Two examples can have the same statements placed on different lines and the test will not flag them.

---
## Issue 3 — free-form comment differs

`isCommentNode()` in `XdocsExamplesAstConsistencyTest` currently strips all single-line comments from the structural AST before comparison. This means two examples can have different free-form comments on different lines or with different text and the test will not flag them.

Only `// violation` and `// ok` marker comments should be exempt — they are intentionally different between examples. All other single-line comments are part of the example's structure and must match.

## Example

```java
// Example1
// section: constructor
MyClass() {}

// Example2
// section: different heading here
MyClass() {}
```

The free-form comment differs between examples but the test passes silently because all `SINGLE_LINE_COMMENT` nodes are discarded.

---

## Impact

All fixes will surface new mismatches in examples that currently pass. Those examples must be either unified (preferred) or added to `SUPPRESSED_EXAMPLES` with justification.

The goal is that extending or reviewing examples requires no manual cross-file consistency check — the test enforces it automatically.

## References

- #18435 (parent issue)
- [PR #20262 comment](https://github.com/checkstyle/checkstyle/pull/20262#issuecomment-4643010103) — properties differ per example
- [PR #19890 comment](https://github.com/checkstyle/checkstyle/pull/19890#issuecomment-4642873546) — new property in overloaded methods check

**Comment by romani** [comment]
There is issue 3:

From https://github.com/checkstyle/checkstyle/pull/20262#issuecomment-4643010103 first sentence.

Comments are ignored, but we need to keep them in sync if there are not ok or violation comment.

**Comment by smita1078** [comment]
> There is issue 3:
> 
> From [#20262 (comment)](https://github.com/checkstyle/checkstyle/pull/20262#issuecomment-4643010103) first sentence.
> 
> Comments are ignored, but we need to keep them in sync if there are not ok or violation comment.

One question, for this do we also consider Javadoc/multiline comments to match? or they should be ignored..

**Comment by stoyanK7** [comment]
> One question, for this do we also consider Javadoc/multiline comments to match? or they should be ignored..

If we enforce the same variable names and the same single-line comments, I don't see why Javadoc/multiline should be exempt. 

@romani Please share your opinion too.

**Comment by stoyanK7** [comment]
> ## Issue 2 — line numbers are not compared

I assumed this is by design. The Examples are limited to 85 chars. Often, it's hard to fit code + `// violation` comment on a single line, so then we need to do `// violation above/below`.

This will force us to rethink some examples and shorten them horizontally (or convert lots of examples to `// ok above/below`/`// violation above/below`)

**Comment by romani** [comment]
Javadoc and multi line comments is part of java code, not a hint single line comments.

Example where we hacked/filled empty space with multi line comment https://github.com/checkstyle/checkstyle/pull/20262#issuecomment-4650445888
But suppose to have empty lines.

**Comment by romani** [comment]
We can change limits of 85 lines .
We can shorten violation messages, we don't have to keep messages verbose.

**Comment by romani** [comment]
Fix is merged
